server/odc-service/src/main/java/com/oceanbase/odc/service/notification/NotificationService.java

```diff
@@ -174,17 +174,18 @@ public Channel updateChannel(@NotNull Long projectId, @NotNull Channel channel)
 
         BaseChannelConfig channelConfig = channel.getChannelConfig();
         if (channelConfig instanceof WebhookChannelConfig && ((WebhookChannelConfig) channelConfig).getSign() == null) {
-            channelPropertyRepository.deleteByChannelIdAndKeyNotEquals(entity.getId(), "sign");
-        } else {
-            channelPropertyRepository.deleteByChannelId(entity.getId());
+            String sign = ((WebhookChannelConfig) channelMapper.fromEntityWithConfig(entity).getChannelConfig())
+                    .getSign();
+            ((WebhookChannelConfig) channelConfig).setSign((sign));
         }
 
         ChannelEntity toBeSaved = channelMapper.toEntity(channel);
-        toBeSaved.setCreatorId(authenticationFacade.currentUserId());
-        toBeSaved.setOrganizationId(authenticationFacade.currentOrganizationId());
-        toBeSaved.setProjectId(projectId);
+        channelRepository.update(toBeSaved);
+
+        channelPropertyRepository.deleteByChannelId(channel.getId());
+        toBeSaved.getProperties().forEach(property -> channelPropertyRepository.save(property));
```

Comment:
may involve performace risk ?

Reply:
Compared to the original @onetonay cascading update, there is no performance rollback. And since the number of properties is in single digits, there should be no performance issues.